### ISSUE TITLE
Error training neus-facto with nerfstudio data parser

### ISSUE THREAD

**Issue by marcostrinca** [nerfstudio-project/nerfstudio#2251]
**Describe the bug**
When training neus-facto with nerfstudio data parser it throws an error:
`KeyError: 'normal_filenames'`

**To Reproduce**
1. process some data using `ns-process-data` with colmap
2. start training process with  `ns-train neus-facto nerfstudio-data --data`

**Expected behavior**
To have my own data trained on neus-facto 

**Screenshots**
![Screen Shot 2023-07-20 at 17 15 00](https://github.com/nerfstudio-project/nerfstudio/assets/759164/b833650e-1b2d-4e15-b1ec-7c3cc018aa0c)

**Comment by tancik** [comment]
`neus-facto` requires normals data which is not provided by the nerfstudio-data processing pipeline. You need to use SDFStudio data.

**Comment by marcostrinca** [comment]
Thanks for your quick reply @tancik 
Sorry. I made a mistake reading #2023. I didn't realize it was for SDFStudio repository only.

Anyway, I couldn't make neus-facto converge using the SDFStudio data parser, even after reading #1732 and processing my data with the mentioned script.
It worthy note that neus-facto works perfectly fine on sdfstudio's demo DTU dataset  with sdfstudio data parser (even removing normals and depth maps).
It seems there's some miscalculations normalizing the cameras on the script. I saw @MvWouden commented something about it and also that he made it work with his own data.

I see two possibilities here:
a) to add a new method on ns-process-data to extract camera poses and create the meta_data.json file instead of transform.json, including generating normal and depth maps using omnidata or mini-omnidata (don't need to resize images do 384x384)
b) customize nerfstudio-data to feed sdfstudio models directly.

In case you guys think it's a good idea I can spend some time working on it. Some guidance would be welcome.

**Comment by MvWouden** [comment]
> Thanks for your quick reply @tancik 
> 
> Sorry. I made a mistake reading #2023. I didn't realize it was for SDFStudio repository only.
> 
> 
> 
> Anyway, I couldn't make neus-facto converge using the SDFStudio data parser, even after reading #1732 and processing my data with the mentioned script.
> 
> It worthy note that neus-facto works perfectly fine on sdfstudio's demo DTU dataset  with sdfstudio data parser (even removing normals and depth maps).
> 
> It seems there's some miscalculations normalizing the cameras on the script. I saw @MvWouden commented something about it and also that he made it work with his own data.
> 
> 
> 
> I see two possibilities here:
> 
> a) to add a new method on ns-process-data to extract camera poses and create the meta_data.json file instead of transform.json, including generating normal and depth maps using omnidata or mini-omnidata (don't need to resize images do 384x384)
> 
> b) customize nerfstudio-data to feed sdfstudio models directly.
> 
> 
> 
> In case you guys think it's a good idea I can spend some time working on it. Some guidance would be welcome.
> 
> 
> 
> 

Just so you know SDFStudio supports nerfstudio-data as input now, so you could also use the SDF-based models on there (neus-facto, bakedsdf, etc).

**Comment by marcostrinca** [comment]
Hi @MvWouden. 
Thanks for your suggestion. 
The nerfstudio version of sdfstudio is outdated and does not support some features I would like to use. But you made me think about to update sdfstudio repo with those features. Not sure what's the best approach, but thinking about this I've just updated the nerfstudio data parser there to support `center_method focus`.
I'll keep working to take more features there for now. 
Thanks guys! Closing this topic.
